### PR TITLE
solution for #transformState

### PR DESCRIPTION
**Pull request by jstmpelowycz**

### CODE REVIEW

**Review by krolich500grn — APPROVED** [review]
good job!

File: src/transformState.js [hunk 1]
```diff
@@ -5,7 +5,25 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  for (const action of actions) {
+    switch (action.type) {
+      case 'clear':
+        Object
+          .keys(state)
+          .forEach(K => delete state[K]);
```
Comment (Ihor-Karpyn) [inline src/transformState.js:13]: Let's use for in

File: src/transformState.js [hunk 2]
```diff
@@ -5,7 +5,25 @@
...
+        break;
+      case 'removeProperties':
+        Object
+          .values(action.keysToRemove)
+          .forEach(K => delete state[K]);
+        break;
```
Comment (Ihor-Karpyn) [inline src/transformState.js:19]: for in

File: src/transformState.js [hunk 3]
```diff
@@ -5,7 +5,25 @@
...
+
+      default:
+        for (const [K, V] of Object.entries(action.extraData)) {
+          state[K] = V;
+        }
```
Comment (Ihor-Karpyn) [inline src/transformState.js:24]: redundat

**Review by Ihor-Karpyn — CHANGES_REQUESTED** [review]

**Review by Ihor-Karpyn — APPROVED** [review]

**Pull request closed without merging**